### PR TITLE
vmui: add functionality to preserve selected columns

### PR DESCRIPTION
**Pull request by Loori-R**
### Describe Your Changes

1) Changed table settings from a popup to a modal window to simplify future functionality additions.
2) Added functionality to save selected columns when data is modified or the page is reloaded. See  #7016.

<details>
  <summary>Example screenshots</summary>
  
demo-1
  
</details>

### Checklist

The following checks are **mandatory**:

- [x] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

**Review by dmitryk-dk — APPROVED** [review]
LGTM!

**Review by hagen1778 — COMMENTED** [review]
Thanks!
The feature for keeping selected columns works great!
But I have questions regarding the `Preserving columns` feature - see https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4780#issuecomment-2360495591

**Comment by Loori-R** [comment]
@hagen1778 , I removed the function for adding custom columns from this pull request.

File: docs/VictoriaLogs/CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ according to [these docs](https://docs.victoriametrics.com/victorialogs/quicksta
 
 ## tip
 
+* FEATURE: [web UI](https://docs.victoriametrics.com/victorialogs/querying/#web-ui): add functionality to preserve selected columns across page reloads. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7016).
```
Comment (hagen1778) [inline docs/VictoriaLogs/CHANGELOG.md:18]:
```suggestion
* FEATURE: [web UI](https://docs.victoriametrics.com/victorialogs/querying/#web-ui): keep selected columns in table view on page reloads. Before, selected columns were reset on each update. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7016).
```

File: docs/changelog/CHANGELOG.md [hunk 2]
```diff
@@ -28,6 +28,7 @@ See also [LTS releases](https://docs.victoriametrics.com/lts-releases/).
...
 * FEATURE: [vmgateway](https://docs.victoriametrics.com/vmgateway/): support parting `vm_access` claims in string format. This is useful for cases when identity provider does not support mapping claims to JSON format.
 * FEATURE: [Single-node VictoriaMetrics](https://docs.victoriametrics.com/) and `vmstorage` in [VictoriaMetrics cluster](https://docs.victoriametrics.com/cluster-victoriametrics/): Adds new metrics for data ingestion: `vm_rows_received_by_storage_total`, `vm_rows_ignored_total{reason="nan_value"}`, `vm_rows_ignored_total{reason="invalid_raw_metric_name"}`, `vm_rows_ignored_total{reason="hourly_limit_exceeded"}`, `vm_rows_ignored_total{reason="daily_limit_exceeded"}`. See this [PR](https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6663) for details.
+* FEATURE: [vmui](https://docs.victoriametrics.com/#vmui): add functionality to preserve selected columns across page reloads. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7016).
```
Comment (hagen1778) [inline docs/changelog/CHANGELOG.md:31]:
```suggestion
* FEATURE: [vmui](https://docs.victoriametrics.com/#vmui): keep selected columns in table view on page reloads. Before, selected columns were reset on each update. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7016).
```

**Review by hagen1778 — APPROVED** [review]

**Review by hagen1778 — APPROVED** [review]
Thanks!

**Comment by hagen1778** [comment]
@Loori-R is it possible to store selected columns in URL?
Currently, page reload causes to reset all selected columns.

**Comment by Loori-R** [comment]
@hagen1778,  Currently, you can check the "Preserve column settings" checkbox in the table settings to save the configuration in localStorage, allowing you to retain settings within the same browser.

We can store column settings in the URL, enabling users to share links with preserved configurations. However, this may significantly increase the URL length, potentially causing issues in certain systems.

**Comment by hagen1778** [comment]
> We can store column settings in the URL, enabling users to share links with preserved configurations. However, this may significantly increase the URL length, potentially causing issues in certain systems.

I don't think this may cause a lot of troubles. But it seems very useful to me, so I can share customized link or put it to alerting rules annotations.

**Comment by Loori-R** [comment]
> I don't think this may cause a lot of troubles. But it seems very useful to me, so I can share customized link or put it to alerting rules annotations. Can we implement it as a new enhancement?

Yes, of course. Will you create the issue, or should I do it?

**Comment by hagen1778** [comment]
See https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7662

**Pull request merged**